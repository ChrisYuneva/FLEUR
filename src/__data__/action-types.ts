export const GOODS = {
    REQUEST: 'GOODS_REQUEST',
    SUCCESS: 'GOODS_SUCCESS',
    FAILURE: 'GOODS_FAILURE',
    SET_GENDER: 'GOODS_SET_GENDER',
    SET_TYPE: 'GOODS_SET_TYPE',
};

export const BASKET = {
    REQUEST: 'BASKET_REQUEST',
    SUCCESS: 'BASKET_SUCCESS',
    FAILURE: 'BASKET_FAILURE',
};
