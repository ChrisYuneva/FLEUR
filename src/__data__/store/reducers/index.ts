import { combineReducers } from 'redux';
import goodsReducer, { GoodsState } from './goods';
import basketReducer, { BasketState } from './basket';

export type AppStore = {
    goods: GoodsState;
    basket: BasketState;
};

export default combineReducers<AppStore>({
    goods: goodsReducer,
    basket: basketReducer,
});
