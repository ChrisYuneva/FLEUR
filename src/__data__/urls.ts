import { getNavigations } from '@ijl/cli';

const navigations = getNavigations('repos');

export const baseUrl = navigations['repos'];
export const URLs = {
    basket: {
        url: navigations['link.repos.basket'],
    },
    home: {
        url: navigations['link.repos.home'],
    },
    home_man: {
        url: navigations['link.repos.home_man'],
    },
    showcase_men: {
        url: navigations['link.repos.showcase_men'],
    },
    showcase_women: {
        url: navigations['link.repos.showcase_women'],
    },
    ordering: {
        url: navigations['link.repos.ordering'],
    },
};
