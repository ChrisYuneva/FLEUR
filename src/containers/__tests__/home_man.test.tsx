import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import HomeMans from '../home_man/index';
import { describe, it, expect } from '@jest/globals';
import { store } from '../../__data__/store';

describe('Тестирование', () => {
    it('Тестируем рендер HomeMans', () => {
        expect.assertions(1);
        const component = mount(
            <Provider store={store}>
                <HomeMans />
            </Provider>
        );

        expect(component).toMatchSnapshot();
    });
});
