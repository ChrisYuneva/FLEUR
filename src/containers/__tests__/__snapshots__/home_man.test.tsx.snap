// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование Тестируем рендер HomeMans 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <HomeMans>
    <div
      className="wrapper"
    >
      <MetaTags />
      <div
        className="header"
      >
        <Link
          className="name"
          to="/home"
        >
          <a
            className="name"
            href="/home"
          >
            FLEUR
          </a>
        </Link>
      </div>
      <div
        className="arrow_div"
      >
        <Link
          className="arrow_d"
          to="/home"
        >
          <a
            className="arrow_d"
            href="/home"
          >
            <img
              className="arrow"
              src="FILE_MOCK"
            />
            <a
              className="arrow_man"
            >
              Женщины
            </a>
          </a>
        </Link>
      </div>
      <Link
        className="home"
        to="/showcase_jackets_men"
      >
        <a
          className="home"
          href="/showcase_jackets_men"
        >
          <div
            className="rectangle"
          >
            <span>
              Перейти к покупкам
            </span>
          </div>
        </a>
      </Link>
    </div>
  </HomeMans>
</Provider>
`;
